plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/handlers/FordiacQuickFixHandler.java

```diff
@@ -91,23 +99,76 @@ public static IMarker[] getMarkersFromEditorSelection(final IStructuredSelection
 				.map(FordiacQuickFixHandler::getMarker).toArray(IMarker[]::new);
 	}
 
+	public static boolean hasMarker(final EditPart editPart) {
+		if (FordiacAnnotationUtil
+				.getAnnotationModel(editPart) instanceof final FordiacMarkerGraphicalAnnotationModel annotationModel) {
+			final var annotation = annotationModel.getAnnotations(editPart.getModel());
+			return !annotation.isEmpty();
+		}
+		return false;
+	}
+
 	public static IMarker getMarker(final EditPart editPart) {
 		if (FordiacAnnotationUtil
 				.getAnnotationModel(editPart) instanceof final FordiacMarkerGraphicalAnnotationModel annotationModel) {
```

Comment:
There should be no need for the instanceof check here.

Reply:
We need the resource of the AnnotationModel for the temporary markers, so I changed it to ResourceMarkerGraphicalAnnotationModel